features/collection/persistence-rrd/src/main/java/org/opennms/netmgt/collection/persistence/rrd/RrdPersistOperationBuilder.java

```diff
@@ -211,7 +222,9 @@ public void commit() throws PersistException {
             List<RrdDataSource> dataSources = getDataSources();
             if (dataSources != null && dataSources.size() > 0) {
                 createRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, getRepository().getStep(), dataSources, getRepository().getRraList(), m_metaData);
-                updateRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, m_timeKeeper.getCurrentTime(), getValues());
+                long rrdTime = Boolean.getBoolean("org.opennms.netmgt.collection.persistence.rrd.useCollectionStartTime")
+                    ? m_collectionTime : m_timeKeeper.getCurrentTime();
```

Comment:
That doesn't seem to fully work.  The method that we care about here is BasePersister::createBuilder(), as that is where we take the resource's timekeeper and make it the builder's timekeeper.  createBuilder() is called by GroupPersister::visitGroup() and OneToOnePersister::visitAttribute(), and in both cases they use the group's/attribute's existing resource as the argument passed to createBuilder().  So modifying what is passed to the visitResource/completeResource calls doesn't do us any good here since neither of those affect what the builder gets.

I'm thinking we need to intercept visitGroup/completeGroup and visitAttribute/completeAttribute too, and in those calls we need to set the group's/attribute's resource in the visitXXX() calls to the wrapper, and then change it back during the completeXXX() calls.  Does that sound right (and safe, specifically setting an resetting the group's/attribute's resource)?